File: 프로그래머스/Level 2/요격 시스템.cpp
// https://school.programmers.co.kr/learn/courses/30/lessons/181188
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> targets) {
    int answer = 1;
    
    sort(targets.begin(), targets.end(), [&](vector<int> &v1, vector<int> &v2) {
		return v1[1] < v2[1];
	});
    
    // for (auto t: targets) cout << t[0] << ' ' << t[1] << '\n';
    
    int cur = targets[0][1];
    for (int i = 1; i < targets.size(); ++i)
    {
        if (cur <= targets[i][0])
        {
            cur = targets[i][1];
            answer++;
        }
    }
    
    return answer;
}

File: 프로그래머스/Level 2/요격 시스템.swift
// https://school.programmers.co.kr/learn/courses/30/lessons/181188
import Foundation

func solution(_ targets:[[Int]]) -> Int {
    var answer: Int = 1
    
    let sortedTargets = targets.sorted { $0[1] < $1[1] }
    //print(sortedTargets)
    var cmp = sortedTargets[0][1]
    
    for target in sortedTargets {
        if cmp <= target[0] {
            answer += 1
            cmp = target[1]
        }
    }
    return answer
}